test/irb/test_ruby_lex.rb

```diff
@@ -632,5 +632,34 @@ def test_unterminated_code
         assert_empty(error_tokens, 'Error tokens must be ignored if there is corresponding non-error token')
       end
     end
+
+    def test_corresponding_token_depth_with_heredoc_and_embdoc
+      codes = [
+        "if true\n  hello\n  p(\n    p(\n    )\n  )",
+        "if true\n  <<~A\n  A\n  p(\n    p(\n    )\n  )",
+        "if true\n=begin\n=end\n  p(\n    p(\n    )\n  )",
+      ]
+      codes.each do |code|
+        lex = RubyLex.new
+        lines = code.lines
+        lex.instance_variable_set('@tokens', RubyLex.ripper_lex_without_warning(code))
+        assert_equal 2, lex.check_corresponding_token_depth(lines, lines.size)
+      end
+    end
+
+    def test_find_prev_spaces_with_multiline_literal
+      lex = RubyLex.new
+      codes = [
+        "if true\n  1\n  hello\n  1\n  world\nend",
```

Comment:
I think to most people it's not easy to newline and spaces in their head. Do you think we can rewrite test cases into something like

```rb
            %[
if true
  1
  hello
  1
  world
end
            ].strip,
```

to improve its readability?

(Note: we can just put `strip` in the `each` block if you think the new format is better)

Reply:
I agree that `"\n  \n  ..."` is not so readable. I rewrite it using `<<~EOC`.